### PR TITLE
Implement a log file for callback timings

### PR DESCRIPTION
**Pull request by pboucher**

### CODE REVIEW

**Comment by neilvictorgrey** [comment]
LGTM.

Tested locally and seems to be working well.

File: src/shotgunEventDaemon.py [hunk 1]
```diff
@@ -961,6 +985,14 @@ def process(self, event):
             if self._stopOnError:
                 self._active = False
 
+        if self._engine.timing_logger:
+            end_time = datetime.datetime.now(SG_TIMEZONE.local)
+            duration = end_time - start_time
+            delay = start_time - event['created_at']
+            msg_format = "event_id=%d created_at=%s callback=%s start=%s end=%s duration=%s error=%s delay=%s"
```
Comment (plongitudes) [inline src/shotgunEventDaemon.py]: i know that this is coming from a different place in the code -- but my only nitpick, and this is my python ignorance, is that i can't figure out why the timestamp has a comma for the decimals place but the rest of the timing info uses a period.

File: src/shotgunEventDaemon.conf.example [hunk 2]
```diff
@@ -36,6 +36,15 @@ logFile: shotgunEventDaemon
...
 
+# Enable Timing logging
+# Timing logging is a separate log file that will log timing information regarding
+# event dispatching and processing run time. This is to help diagnose which plugins
+# are taking the most amount of time and where any potential queue processing
+# delay might be coming from. Valid values are `on` or to enable or anything else
+# to disable.
+# timing_log: on
+timing_log: off
+
```
[the remark this answers is not in the document]
Comment (pboucher) [reply src/shotgunEventDaemon.conf.example]: I'll document in the wiki.

File: src/shotgunEventDaemon.py (hunk 1 above)
Comment (pboucher) [reply src/shotgunEventDaemon.py]: This is the logging Formatter using `asctime`. I probably should have used ISO but at this point, I don't want to break compatibility is people are parsing it to throw logs in some aggregation system.

The datetimes in the message use ISO format.

I've modified the output format of the durations to properly represent days if present in a more parseable format.

**Pull request merged**